### PR TITLE
Add new helm chart to install redis-cluster in incubator repository

### LINKED ISSUE #12424: Add helm-chart for redis-cluster
**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):

FEATURE REQUEST:
- Add a chart for installing redis as a redis-cluster

**What is a redis-cluster?**

Redis Cluster is a set of Redis instances working together to make your data storage scale across nodes while also making it a bit more resilient. Data is automatically split across nodes and it supports a master/slave setup for increased availability in case of a failure. It is basically a data sharding strategy, with the ability to reshard keys from one node to another while the cluster is running, together with a failover procedure that makes sure the system is able to survive certain kinds of failures.

For Reference: https://redislabs.com/redis-features/redis-cluster

### PR DESCRIPTION
**Pull request by vidhig**
Signed-off-by: vidhig <vidhigupta0494@gmail.com>

#### What this PR does / why we need it:
This PR will allow installing redis in cluster-mode.

#### Which issue this PR fixes
  - fixes [#12424](https://github.com/helm/charts/issues/12424) 

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by vidhig** [comment]
The lint-charts on the CI is failing, but when I run `helm lint .` on local it is passing.
```
==> Linting .
Lint OK

1 chart(s) linted, no failures```

**Comment by vidhig** [comment]
The lint-charts test failed on the CI, but when I run `helm lint .` on my system, it passes.
```
==> Linting .
Lint OK

1 chart(s) linted, no failures
```

File: incubator/redis-cluster/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+icon: https://bitnami.com/assets/stacks/redis/img/redis-stack-220x234.png
+version: 0.1.0
+appVersion: 5.0
+description: Redis is an open source, advanced key-value store. It is often referred to as a data structure server since keys can contain strings, hashes, lists, sets and sorted sets. This chart will allow you to install a redis cluster which allows the ability to reshard keys from one node to another while the cluster is running, together with a failover procedure that makes sure the system is able to survive certain kinds of failures.
+keywords:
+- redis-cluster
+- keyvalue
+home: http://redis.io/
+maintainers:
+  - name: 'Vidhi Gupta'
```
Comment (neglect-yp) [inline incubator/redis-cluster/Chart.yaml]: >  ✖︎ incubator/redis-cluster > Error validating maintainer 'Vidhi Gupta': 404 Not Found

It probably should be a valid username of GitHub (not display name).

Comment (vidhig) [reply incubator/redis-cluster/Chart.yaml]: yeah, thank you @neglect-yp 
I had seen the error and fixed this already.

**Comment by vidhig** [comment]
Hello @cpanato and @lachie83

Do you have any feedback on this PR?
Am I missing something as one check is in `pending` state and one other in `exepcted`?

Thank you

**Comment by maorfr** [comment]
hi @vidhig 

Have you considered hosting this chart and adding it to helm hub?
https://github.com/helm/hub/blob/master/Repositories.md

**Comment by vidhig** [comment]
@maorfr Thank you for the suggestion, I will look into it.

**Comment by vidhig** [comment]
> hi @vidhig
> 
> Have you considered hosting this chart and adding it to helm hub?
> https://github.com/helm/hub/blob/master/Repositories.md

@maorfr Thank you for the suggestion, I will look into it.

**Pull request closed without merging**